tests/Routing/RoutingUrlGeneratorTest.php

```diff
@@ -115,6 +115,64 @@ public function testBasicRouteGeneration()
 	}
 
 
+	public function testBasicRouteGenerationOptionalParameterMatch()
+	{
+		$url = new UrlGenerator(
+				$routes = new Illuminate\Routing\RouteCollection,
+				$request = Illuminate\Http\Request::create('http://www.foo.com/')
+		);
+
+		$route = new Illuminate\Routing\Route(array('GET'), 'foo/bar/{baz?}', array('as' => 'foobaz'));
+		$routes->add($route);
+
+		$this->assertEquals('http://www.foo.com/foo/bar/taylor', $url->route('foobaz', ['baz' => 'taylor']));
+	}
+
+
+	public function testBasicRouteGenerationOptionalParameterNoMatch()
+	{
+		$url = new UrlGenerator(
+			$routes = new Illuminate\Routing\RouteCollection,
+			$request = Illuminate\Http\Request::create('http://www.foo.com/')
+		);
+
+		$route = new Illuminate\Routing\Route(array('GET'), 'foo/bar/{baz?}', array('as' => 'foobaz'));
+		$routes->add($route);
+
+		$this->assertEquals('http://www.foo.com/foo/bar?taylor=otwell', $url->route('foobaz', ['taylor' => 'otwell']));
+	}
+
+
+	public function testBasicRouteGenerationOptionalParameterNoMatchIntegerKeyValue()
+	{
+		$url = new UrlGenerator(
+				$routes = new Illuminate\Routing\RouteCollection,
+				$request = Illuminate\Http\Request::create('http://www.foo.com/')
+		);
+
+		$route = new Illuminate\Routing\Route(array('GET'), 'foo/bar/{baz?}', array('as' => 'foobaz'));
+		$routes->add($route);
+
+		// or /foo/bar?1&2 ???
+		$this->assertEquals('http://www.foo.com/foo/bar//1/2', $url->route('foobaz', [1,2]));
```

Comment:
There wouldn't be a way to differentiate them really.

On Fri, Mar 27, 2015 at 4:40 PM, Aimeos <notifications@github.com> wrote:

> In tests/Routing/RoutingUrlGeneratorTest.php
> <https://github.com/laravel/framework/pull/8198#discussion_r27333157>:
>
> > +		$this->assertEquals('http://www.foo.com/foo/bar?taylor=otwell', $url->route('foobaz', ['taylor' => 'otwell']));
> > +	}
> > +
> > +
> > +	public function testBasicRouteGenerationOptionalParameterNoMatchIntegerKeyValue()
> > +	{
> > +		$url = new UrlGenerator(
> > +				$routes = new Illuminate\Routing\RouteCollection,
> > +				$request = Illuminate\Http\Request::create('http://www.foo.com/')
> > +		);
> > +
> > +		$route = new Illuminate\Routing\Route(array('GET'), 'foo/bar/{baz?}', array('as' => 'foobaz'));
> > +		$routes->add($route);
> > +
> > +		// or /foo/bar?1&2 ???
> > +		$this->assertEquals('http://www.foo.com/foo/bar//1/2', $url->route('foobaz', [1,2]));
>
> @taylorotwell <https://github.com/taylorotwell> @GrahamCampbell
> <https://github.com/GrahamCampbell> I'm not a friend of such solutions
> but maybe it's then the only possibility to have different generating code
> for Route::controller routes and the other ones.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laravel/framework/pull/8198/files#r27333157>.
>